pennylane_qiskit/qiskit_device2.py

```diff
@@ -98,7 +98,31 @@ def circuit(x):
     """
     # Code to acquire session:
     existing_session = device._session
-    session = Session(backend=device.backend)
+    session_options = {
+        "service": device.service,
+        "backend": device.backend,
+    }
+
+    for k, v in kwargs.items():
+        # Options like service and backend should be tied to the settings set on device
+        if k in session_options and k != "max_time":
+            warnings.warn(f"Using '{k}' set in device, {device.backend}", UserWarning)
+
+        # Need "_" since `max_time` attribute on Session is `_max_time`
+        # When there is overlap between the Session options on the device and the session options
+        # passed in via qiskit_session, we prefer the ones passed in via qiskit_session
+        elif existing_session and (
+            hasattr(existing_session, "_" + k) or hasattr(existing_session, k)
+        ):
+            warnings.warn(
+                f"`{k}` was set in the Session passed to the device. Using `{k}` '{v}' set in `qiskit_session`.",
+                UserWarning,
+            )
```

Comment:
To me this feels like it implies that if I *don't* pass a value for `max_time`, and my initial session did have a value for `max_time`, then my initial value will be used, since I'm notified that its being overwritten - as if its stored somewhere and will remain the same until I change it. But that's not true, right? The default value will just be used?

Reply:
In the case where:
1. There is an initial session with some value for `max_time` passed to the device.
2. And then you run `qiskit_session(dev)` without setting `max_time`. 

Would result in
1. Don't get any warnings
2. But the session will use `None` for `max_time`, which Qiskit in turn interprets as some default value.
3. When you close the session and run the circuit again it will use the max_time stored on the device.

I can push a fix to use the `max_time` set in the initial session (i.e. don't override with `None`). Alternatively, we could raise a warning letting them know this is going to be ran with whatever Qiskit set as the default...

Reply:
It's not so much that I would expect it to use the `max_time` from the initial session. It's more that I *wouldn't* expect that, but the warning to me implies that that's what happens, which I think I would find confusing.

I see that we want to get `service` and `backend` from the device, so raising a warning there if they try to change it I understand. But if I set `max_time` the first time and the second time to separate values, I think those are two independent things - I would find a warning for that, as if something undefined or unexpected is happening, confusing.

What if we just assume that all sessions are related to the device, and must have a `backend` and `service` from the device, but they aren't related to each other and so we don't compare them at all?

Reply:
> When there are overlapping settings in the Session passed in on device initialization and the qiskit_session, we will raise a warning letting them know this is strange, and explain what exactly happened.

Basically this. Imagining I'm a user - I'm surprised that it would be strange to use a context manager to temporarily use a session with different settings, since that seems to be what the context manager is provided for. So when a warning is raised to let me know it's strange, I feel confused about what I did wrong, and worried I've misunderstood something.

Reply:
Ah, I see what you mean now. Yes that is a bit of a strange warning now that I think about it. I will remove the warning for `max_time` but maybe it is useful to keep it in for the `service` and `backend`?

Reply:
>maybe it is useful to keep it in for the service and backend?

Agreed! For `service` and `backend` the user is passing something unexpected that we will overwrite, so a warning seems good to me